.github/workflows/mdbook_ci.yml

```diff
@@ -35,17 +39,13 @@ jobs:
         run: mdbook build
 
       - name: Create tarball 📦
+        if: ${{ github.ref == 'refs/heads/main' }}
         run: tar -cvf spec_mdbook.tar ./book
 
       - name: Create mdbook build artifact 📦
+        if: ${{ github.ref == 'refs/heads/main' }}
         uses: actions/upload-artifact@v3
         with:
-          name: spec_mdbook
+          name: spec_mdbook-${{ github.sha }}
           path: spec_mdbook.tar
 
```

Comment:
@wilwade  this will tag the artifact on main with its hash to during build time we can get the sha on commit for which tag was created and only download those artifacts and not the latest one

I think this should work but will know more once I merge to main